### PR TITLE
fix(ios): use correct iOS destination platform during build

### PR DESCRIPTION
**Pull request by lionelbouth**
https://github.com/appcelerator/titanium_mobile/discussions/12602

https://github.com/appcelerator/titanium_mobile/issues/13053

JIRA: https://jira.appcelerator.org/browse/TIMOB-28545

### CODE REVIEW

**Comment by m1ga** [comment]
awesome, @lionelbouth :+1: 
You could change your title to: `fix(ios): correct iOS destination platform during build` or something like that so it will look nicer.

**Comment by ewanharris** [comment]
Thanks @lionelbouth! It looks like this maybe caused the `macos` target to fail on a build. We'll investigate and push any required fixes to your branch

**Comment by ewanharris** [comment]
@lionelbouth I just pushed a commit to only apply the `-destination generic/platform=iOS` when targeting the `device` target type which fixes the `macos` target issue. Could you try applying that change and see if it works for you on an M1 mac?

**Comment by lionelbouth** [comment]
Tried your commit on 9.2.1.GA,  on my MAC M1 : not working.

**Comment by ewanharris** [comment]
@lionelbouth when you say not working, what are you targeting (device or another target?) and what is the error you are receiving?

**Comment by lionelbouth** [comment]
Sorry my bad.
I tried to build ipa whith Ad hoc provisioning profiles ...
IMO, the correct answer should be **if( this.target ==='device' ||  this.target ==='dist-adhoc' )**

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -7102,6 +7102,8 @@ iOSBuilder.prototype.invokeXcodeBuild = async function invokeXcodeBuild(next) {
 			this.logger.warn(`The app is using native modules (${Array.from(this.legacyModules)}) that do not support arm64 simulators, we will exclude arm64. This may fail if you're on an arm64 Apple Silicon device.`);
 			args.push('EXCLUDED_ARCHS=arm64');
 		}
+	} else if ( this.target === 'device' || this.target === 'dist-adhoc' ){
```
Comment (ewanharris) [inline iphone/cli/commands/_build.js:7105]: Does this also need to include the `dist-appstore` target or is the error not seen when targeting that?

Comment (ewanharris) [reply iphone/cli/commands/_build.js:7105]: Also, eslint will complain about the spaces in the brackets. If you have this checked out locally you can run `npm run format:js` to fix them and any other issues automatically, you might need to run `npm ci` first.

Comment (lionelbouth) [reply iphone/cli/commands/_build.js:7105]: No need to include dist-appstore. 
Probably because appstore provisioning profiles don't include devices

**Comment by m1ga** [comment]
btw this also fixed
```

  --- xcodebuild: WARNING: Using the first of multiple matching destinations:
    { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
    { platform:macOS, arch:arm64, variant:Designed for [iPad,iPhone], id:00008103-001444810A31001E }
    { platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
    { platform:iOS Simulator, id:EDAD5055-209E-43BF-A48F-F0A2D8D26F43, OS:15.0, name:iPad (9th generation) }
    { platform:iOS Simulator, id:6D279D6B-91AD-43A0-ADEA-9C19BDE7F867, OS:15.0, name:iPad Air (4th generation) }
    { platform:iOS Simulator, id:CB7171EA-2225-4747-9F52-701A93AAEB11, OS:15.0, name:iPad Pro (9.7-inch) }
    { platform:iOS Simulator, id:5142D23C-2C4B-4A12-B1D7-0F50CC53106F, OS:15.0, name:iPad Pro (11-inch) (3rd generation) }
    ....

```
for me which suddenly appeared. Not sure why, running Xcode 13 for a while now and using an old Macbook Air

**Comment by sgtcoolguy** [comment]
I can confirm this fixes #12602 for me on my M1 Mac mini. Building for device would fail before complaining the host machine was not on the provisioning profile. With this fix, it no longer does.

**Review by ewanharris — APPROVED** [review]
CR Pass!

Thanks @lionelbouth!

**Pull request merged**